Mathlib/RingTheory/AdicCompletion/Algebra.lean

```diff
@@ -59,11 +64,35 @@ def subalgebra : Subalgebra R (∀ n, R ⧸ (I ^ n • ⊤ : Ideal R)) :=
 def subring : Subring (∀ n, R ⧸ (I ^ n • ⊤ : Ideal R)) :=
   Subalgebra.toSubring (subalgebra I)
 
-instance : CommRing (AdicCompletion I R) :=
-  inferInstanceAs <| CommRing (subring I)
-
-instance : Algebra R (AdicCompletion I R) :=
-  inferInstanceAs <| Algebra R (subalgebra I)
+@[irreducible]
```

Comment:
I am not sure if the `irreducible` attribute does anything on an instance. The pattern is 
```suggestion
@[irreducible]
def mul (x y : AdicCompletion I R) : AdicCompletion I R := ⟨x.val * y.val, by simp [x.property, y.property]⟩

instance instMul : Mul (AdicCompletion I R) where 
  mul := mul 
```
so Lean can never unfold `mul`

Reply:
Should I do this for every data field? Also `zero`, `one`, etc.?

Reply:
I think it is a worth a test.